runtime/reconcilers/project_parser.go

```diff
@@ -685,6 +689,10 @@ func equalSourceSpec(a, b *runtimev1.SourceSpec) bool {
 }
 
 func equalModelSpec(a, b *runtimev1.ModelSpec) bool {
+	if a.InputConnector == "local_file" {
+		// hack to make sure that models with `local_file` as input connector are always reconciled
+		return false
+	}
```

Comment:
Could this hack be avoided by moving the computation of the hash into the parser instead of in the executor? So the parser would compute the hash and set `inputProps["filesHash"]`.

Reply:
Yes. Good idea.